pkg/integrations/manager.go

```diff
@@ -220,9 +222,14 @@ func (m *Manager) ApplyConfig(cfg ManagerConfig) error {
 				continue
 			}
 			p.stop()
+			m.im.DeleteConfig(key)
 			delete(m.integrations, key)
 		}
 
+		if !ic.CommonConfig().Enabled {
+			continue
+		}
```

Comment:
I still think a function to return a slice of enabled integrations would be easier to follow than having to maintain this check across the two loops, IMO.

Reply:
We still need the integrations in the array to disable/remove them, ie going from enabled -> disabled on reload.